src/averageAges.js

```diff
@@ -15,11 +15,13 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const men = people.filter(person => person.sex === 'm'
+  && (!century || Math.ceil(person.died / 100) === century));
+
+  const ages = men.map(person => person.died - person.born);
+  const totalAge = ages.reduce((sum, age) => sum + age, 0);
+
+  return totalAge / men.length;
```

Comment:
You've repeated the same logic for calculating average age for array of people several time
Let's create separate function with this functionality